cmd/gardener-controller-manager/app/app.go

```diff
@@ -216,6 +217,41 @@ func run(ctx context.Context, log logr.Logger, cfg *controllermanagerconfigv1alp
 			}
 		}
 
+		managedSeeds := &seedmanagementv1alpha1.ManagedSeedList{}
+		if err := mgr.GetClient().List(ctx, managedSeeds); err != nil {
+			return fmt.Errorf("failed listing objects: %w", err)
+		}
+		if err := meta.EachListItem(managedSeeds, func(o runtime.Object) error {
+			fns = append(fns, func(ctx context.Context) error {
+				obj := o.(client.Object)
+
+				logger := mgr.GetLogger().WithValues("objectKey", client.ObjectKeyFromObject(obj))
+				logger.Info("Check the seed name label for itself")
+				label := v1beta1constants.LabelPrefixSeedName + obj.GetName()
+				logger = logger.WithValues("label", label)
+				if _, ok := obj.GetLabels()[label]; ok {
+					logger.Info("Label exists, send an empty patch to the ManagedSeed so that the mutating webhook can remove the superfluous seed name label")
+					emptyPatch := client.MergeFrom(obj)
+					if err := mgr.GetClient().Patch(ctx, obj, emptyPatch); err != nil {
+						return fmt.Errorf("failed to patch managed seed %s: %w", client.ObjectKeyFromObject(obj), err)
+					}
+
+					// assert the mutating webhook runs on the correct version and removed the label
+					managedSeed := &seedmanagementv1alpha1.ManagedSeed{}
+					if err := mgr.GetClient().Get(ctx, client.ObjectKey{Name: obj.GetName()}, managedSeed); err != nil {
+						return fmt.Errorf("failed to get managed seed %s: %w", client.ObjectKeyFromObject(obj), err)
+					} else if _, ok := managedSeed.GetLabels()[label]; ok {
+						return fmt.Errorf("the label %s on the managed seed %s is still present, the mutating webhook is running in an older version", label, client.ObjectKeyFromObject(obj))
+					}
```

Comment:
You do not need to read the object again - after sending the empty patch, the `managedSeed` object is already updated with whatever the API server returned. Hence, you can just check if the label is still present after the empty patch.